### DISCUSSION TITLE
Wireless/ethernet RS485 interface

### DISCUSSION THREAD

**Discussion by jjarboe01** [aqualinkd/AqualinkD#293] [Ideas]
Some background first - I hate relying on any company's cloud based solutions, especially when there is a GROSS lack of support on their API's.  I run Home Assistant server as VM, and look to integrate everything I can to be local control within that system to reduce cloud dependency, and keep my smart home smart during any internet outage or cloud outage.

I am working on using your code to replace the GARBAGE iAquaLink app with something that will integrate with HomeAssistant.  My problem has been the salt cell having issues and not knowing until the pool turns green because Jandy will not expose the AquaPure system in their API's.

My goal is to use your code set as the base to connect to this (https://www.amazon.com/dp/B08GQ78821?psc=1&smid=A28ZWXW3ZSVNZU&ref_=chk_typ_imgToDp) device - I use one on my hot tub for a HomeAssistant integration.  From there, bring the devices into HomeAssistant, rebuild my dashboard for the pool to be functional with this new integration, and finally set up automations for alerts on the salt system to know right away what is going on.

I have to figure out how to get your software to recognize this device over the network to function.  Once I figure this out (device will be here tomorrow from Amazon), I will look to take your code base and create a docker install so this can be deployed through HomeAssistant, rather than need a full linux distro install.  Ultimately, my goal would be to be able to take your code base and use it as a better integration for HA for Jandy so that you have full control, and package the alerts for certain conditions to be deployed when you install the plugin.  Hopefully, these efforts will be beneficial for others in the home automation space who are stuck with these Jandy devices!  

Also looking for BS reasons to get back to coding and DevOps approach for things around my house!

**Comment by bcre3306** [comment]
Check here [336](https://github.com/sfeakes/AqualinkD/discussions/336)
